test/common/testlib.py

```diff
@@ -767,12 +779,42 @@ def snapshot(self, title, label=None):
             attach(filename, move=True)
             print("Wrote HTML dump to " + filename)
 
-    def assert_pixels(self, selector, key, ignore=[]):
-        """Compare the given element with its reference"""
+    def _set_window_size(self, width, height):
+        self.cdp.invoke("Emulation.setDeviceMetricsOverride",
+                        width=width, height=height,
+                        deviceScaleFactor=0, mobile=False)
+
+    def set_layout(self, name):
+        layout = next(filter(lambda l: l["name"] == name, self.layouts))
```

Comment:
The next() here looks really strange.. It's python syntax for getting the first element of an iterator, but it *looks* like an off-by-one.

Perhaps this is a bit clearer:

    [layout for layout in self.layouts if layout["name"] == name][0]

The failure mode is a bit different -- your's would just throw a `StopIteration`, mine would throw an `IndexError`, so in either case this should be caught and replaced with something more readable. Or you drop the `[0]` and check for an empty list.

But what you should *really* be doing is to convert this array to a dict, indexed by `name` :grin:

Reply:
It started out as a dict... :-)  I turned it into an array because:

- `self.current_layout` now points to the actual dict (instead of containing the name of a layout)
- which means the dict needs a "name" or "label" field for forming the filenames
- which would mean repeating the name: `{ "mobile": { "label": "mobile", ... } }`
- also, with an array, the initial layout can just be the first. With a dict, we would have to hard-code "desktop" as the default, or add some field like "is_default".

Personally, *shrug*  I was surprised that there is no `find` in Python and that `next` is considered canonical, but *shrug*